### PR TITLE
Allow allocating clump models

### PR DESCRIPTION
**Pull request by TheNormalnij**
This fixes issues with missing animations on custom models

```lua
engineRequestModel( 'clump', parentId )
```

You can replace a new model using `engineReplaceModel` or IMG functions.

Test resource:
[clump_model_test.zip](https://github.com/multitheftauto/mtasa-blue/files/12165405/clump_model_test.zip)

### CODE REVIEW

File: Client/game_sa/CModelInfoSA.cpp [hunk 1]
```diff
@@ -1710,6 +1710,22 @@ void CModelInfoSA::MakeTimedObjectModel(ushort usBaseID)
     CopyStreamingInfoFromModel(usBaseID);
 }
 
+void CModelInfoSA::MakeClumpModel(ushort usBaseID)
+{
+    CClumpModelInfoSAInterface* m_pInterface = new CClumpModelInfoSAInterface();
```
Comment (tederis) [inline Client/game_sa/CModelInfoSA.cpp]: **m_** prefix should correspond only to class members

**Comment by CrosRoad95** [comment]
if newly allocated models replaced with clump model has all atomics visible then this pull request in my opinion should be merged

File: Client/game_sa/CModelInfoSA.cpp (hunk 1 above)
Comment (TheNormalnij) [reply Client/game_sa/CModelInfoSA.cpp]: I  agree with you. It's copy/pasted from bad code

File: Client/game_sa/CModelInfoSA.h [hunk 2]
```diff
@@ -243,6 +243,16 @@ struct CTimeInfoSAInterface
     short m_wOtherTimeModel;
 };
 
+class CClumpModelInfoSAInterface : public CBaseModelInfoSAInterface
+{
+public:
+    union
+    {
+        char*  m_animFileName;
+        uint32 m_nAnimFileIndex;
```
Comment (botder) [inline Client/game_sa/CModelInfoSA.h]:
```suggestion
        char*    m_animFileName;
        uint32_t m_nAnimFileIndex;
```

**Review by botder — CHANGES_REQUESTED** [review]

**Review by botder — APPROVED** [review]

**Pull request merged**